csep/utils/plots.py

```diff
@@ -693,18 +701,30 @@ def plot_catalog(catalog, ax=None, show=False, extent=None, set_global=False, pl
         dv = (bbox[3] - bbox[2]) / 20.
         extent = [bbox[0] - dh, bbox[1]+dh, bbox[2] -dv, bbox[3] + dv]
 
+    apprx = False
+    central_latitude = 0.0
+    if projection == 'fast':
+        projection = ccrs.PlateCarree()
+        apprx = True
+        n_lats = len(catalog.region.ys) // 2
+        central_latitude = catalog.region.ys[n_lats]
+
     # Instantiage GeoAxes object
     if ax is None:
         fig = pyplot.figure(figsize=figsize)
         ax = fig.add_subplot(111, projection=projection)
-        if set_global:
-            ax.set_global()
-        else:
-            ax.set_extent(extents=extent, crs=ccrs.PlateCarree()) # Defined extent always in lat/lon
+    else:
+        fig = ax.get_figure()
+
+    if set_global:
+        ax.set_global()
+    else:
+        ax.set_extent(extents=extent, crs=ccrs.PlateCarree())  # Defined extent always in lat/lon
```

Comment:
should the `crs` here be `projection` and not `ccrs.PlateCarree()`

Reply:
According to the docs: "_Set the extent (x0, x1, y0, y1) of the map in the **given coordinate system**._ [[scitools.org.uk/cartopy/docs/...](https://scitools.org.uk/cartopy/docs/v0.13/matplotlib/geoaxes.html#cartopy.mpl.geoaxes.GeoAxes.set_extent)]. By default (`crs=None`), it would take over the projection of the axis (i.e. `projection`), but since the extent will be - in our case - (always?) specified in the same coordinate system as the original data (i.e. PlateCaree()), it is good as is.

See also [stackoverflow.com](https://stackoverflow.com/a/43505490).